File: client/src/app/_models/userParams.ts
import { User } from "./user";

export class UserParams {
    gender: string;
    minAge = 18;
    maxAge = 99;
    pageNumber = 1;
    pageSize =10;
    status= 'Active';
    type= 'Employee'
    orderBy = 'lastActive';
    groups = 'Team';


    constructor(user: User) {
       this.gender = user.gender === 'female' ? 'male' : 'female'
    }

}

File: API/Helpers/UserParams.cs
namespace API.Helpers
{
    public class UserParams : PaginationParams
    {
     
        public string CurrentUsername { get; set; }
        public string Gender { get; set; }
        public int MinAge { get; set; } = 18;
        public int MaxAge { get; set; } = 100;
        public string Status { get; set; } = "Active";
        public string Groups { get; set; } = "Team";
        public string Type { get; set; } = "Employee";
        public string OrderBy { get; set; } = "lastActive";

    }
}